### PR TITLE
Implement jax.numpy.argpartition

### LINKED ISSUE #2079: Implement NumPy sorting routines
xref https://docs.scipy.org/doc/numpy/reference/routines.sort.html

Extracted from #70:

 - [x] np.sort
 - [x] np.lexsort
 - [x] np.argsort
 - [x] np.msort
 - [x] np.sort_complex
 - [x] np.partition
 - [ ] np.argpartition

All of these should be relatively doable, though it's worth noting that I don't think XLA currently has efficient partitioning schemes.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes https://github.com/google/jax/issues/2079

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -3530,10 +3530,13 @@ def msort(a):
...
 @_wraps(np.partition, lax_description="""
 The jax version requires the ``kth`` argument to be a static integer rather than
-a general array. This is implemented via two calls to :func:`jax.lax.top_k`.
+a general array. This is implemented via two calls to :func:`jax.lax.top_k`. If
+you're only accessing the top or bottom k values of the output, it may be more
+efficient to call :func:`jax.lax.top_k` directly.
 """)
 @partial(jit, static_argnames=['kth', 'axis'])
 def partition(a: ArrayLike, kth: int, axis: int = -1) -> Array:
+  # TODO(jakevdp): handle NaN values like numpy.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3539]: Should we document this difference in behavior?

**Review by mattjj — APPROVED** [review]
Nice!

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:3542]: Good idea - done!

**Comment by jakevdp** [comment]
With this I think we can close Issue #70!!

**Pull request merged**